### PR TITLE
Moved Criteria, SortClauses and CriterionMapper form Product Catalog package

### PR DESCRIPTION
**Pull request by mikadamczyk**
| :ticket: Issue | N/A |
|----------------|-----------|

### CODE REVIEW

File: src/bundle/Resources/config/services/persistence.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+services:
```
Comment (konradoboza) [inline src/bundle/Resources/config/services/persistence.yaml]: Shouldn't this config reside in some more meaningful file like: `services/search.yaml` or `services/query.yaml`?

File: src/contracts/Values/Query/SortClause/FieldValueSortClause.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\CoreSearch\Values\Query\SortClause;
+
+use Ibexa\Contracts\CoreSearch\Values\Query\AbstractSortClause;
+
+class FieldValueSortClause extends AbstractSortClause
```
Comment (konradoboza) [inline src/contracts/Values/Query/SortClause/FieldValueSortClause.php]: Could it be `final`?

File: src/contracts/Values/Query/Criterion/LogicalOr.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\CoreSearch\Values\Query\Criterion;
+
+class LogicalOr extends AbstractCompositeCriterion
```
Comment (konradoboza) [inline src/contracts/Values/Query/Criterion/LogicalOr.php]: Could it be `final`?

File: src/contracts/Values/Query/Criterion/LogicalAnd.php [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\CoreSearch\Values\Query\Criterion;
+
+class LogicalAnd extends AbstractCompositeCriterion
```
Comment (konradoboza) [inline src/contracts/Values/Query/Criterion/LogicalAnd.php]: Could it be `final`?

File: src/contracts/Values/Query/Criterion/FieldValueCriterion.php [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\CoreSearch\Values\Query\Criterion;
+
+class FieldValueCriterion implements CriterionInterface
```
Comment (konradoboza) [inline src/contracts/Values/Query/Criterion/FieldValueCriterion.php]: Could it be `final`?

**Review by konradoboza — APPROVED** [review]

**Review by adamwojs — APPROVED** [review]

**Comment by adamwojs** [comment]
You can merge it up @mikadamczyk

**Pull request merged**